pkgs/development/compilers/circt/circt-llvm.nix

```
@@ -0,0 +1,57 @@
+{ stdenv
+, cmake
+, ninja
+, circt
+, llvm
+, python3
+}: stdenv.mkDerivation {
+  pname = circt.pname + "-llvm";
+  version = circt.version;
+  src = circt.src + /llvm;
+
+  requiredSystemFeatures = [ "big-parallel" ];
+
+  nativeBuildInputs = [ cmake ninja python3 ];
+
+  cmakeDir = "../llvm";
```

```suggestion
  preConfigure = ''
    cd llvm
  '';
```

This seems to be the current best practice treewide.

---

Thanks for the suggestion!
It looks like it needs to be like this. 

```nix
preConfigure = ''
  cd llvm/llvm
'';
```

I am currentry checking if this builds correctly.

BTW, I found that you could do a similar thing using `sourceRoot`. Do you have any opinions on that?

```nix
sourceRoot = "source/llvm/llvm";
```

---

`sourceRoot` seems to be frowned upon. Since it depends on some internal behaviors of nix (like naming the source folder `source`) and also affected by some other issues that I cannot remember. `cd` is more predictable and reliable.

---

I see. Thanks!
I'll update this PR when the build finishes.